### PR TITLE
SKALE-5032 add message and new deposit box

### PR DESCRIPTION
**Pull request by payvint**

### CODE REVIEW

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by kladkogex — DISMISSED** [review]

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by kladkogex — DISMISSED** [review]

File: proxy/contracts/Messages.sol [hunk 1]
```diff
@@ -119,6 +121,17 @@ library Messages {
         uint256 tokenId;
     }
 
+    /**
+     * @dev Structure for describing base ERC721 with metadata.
+     */
+    struct TransferErc721MessageWithMetadata {
```
Comment (DimaStebaev) [inline proxy/contracts/Messages.sol:127]: `TransferErc721MessageWithMetadata` should be inherited from `TransferErc721Message` but not `BaseMessage`.

File: proxy/scripts/test_upgrade.sh [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
 DEPLOYED_TAG="$(cat "$GITHUB_WORKSPACE"/proxy/DEPLOYED)"
+VERSION_TAG="$(cat "$GITHUB_WORKSPACE"/VERSION)"
 DEPLOYED_VERSION="$(echo "$DEPLOYED_TAG" | cut -d '-' -f 1)"
 DEPLOYED_DIR=$GITHUB_WORKSPACE/deployed-proxy/
 
 
 git clone --branch "$DEPLOYED_TAG" "https://github.com/$GITHUB_REPOSITORY.git" "$DEPLOYED_DIR"
 
-npx ganache-cli --gasLimit 9000000 --quiet --allowUnlimitedContractSize &
```
Comment (DimaStebaev) [inline proxy/scripts/test_upgrade.sh]: Does it not fit to 9M?

File: proxy/contracts/Messages.sol (hunk 1 above)
Comment (payvint) [reply proxy/contracts/Messages.sol:127]: Good point!

File: proxy/scripts/test_upgrade.sh (hunk 2 above)
Comment (payvint) [reply proxy/scripts/test_upgrade.sh]: Unexpectedly changed - thought it was a gas limit issue - but it is not - can downgrade it

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by sergiy-skalelabs — APPROVED** [review]

**Review by DimaStebaev — APPROVED** [review]

**Pull request merged**